pkg/ottl/ottlfuncs/func_scale.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ScaleArguments[K any] struct {
+	Value      ottl.GetSetter[K]
```

Comment:
The metric Editors in the transform processor use the `GetMetric` method on `ottlmetric.TransformContext` to get the metric, I think we could take advantage of that to cut out this parameter.

Reply:
Agreed, we shouldn't need to pass in a target, the target will implicitly be all datapoints in the current metric.